packages/code/src/types.ts

```diff
@@ -33,19 +35,46 @@ export interface SyntaxProps extends React.HTMLAttributes<HTMLElement> {
   /**
    * Determines whether or not the syntax will be rendered in dark mode.
    *
-   * @default: `false`
+   * default: `false`
    */
   darkMode?: boolean;
 
   /**
    * Shows line numbers. This is specifically used for the Code component implementation.
    *
-   * @default: `false`
+   * default: `false`
    */
   showLineNumbers?: boolean;
 
   /**
    * An array of lines to highlight. The array can only contain numbers corresponding to the line numbers to highlight, and / or tuples representing a range (e.g. `[6, 10]`);
    */
-  highlightLines?: Array<number | [number, number]>;
+  highlightLines?: LineHighlightingDefinition;
+}
+
+export interface CodeProps extends Omit<SyntaxProps, 'onCopy'> {
+  /**
+   * Shows window chrome for code block;
+   *
+   * default: `false`
+   */
+  showWindowChrome?: boolean;
+
+  /**
+   * Renders a file name or other descriptor for a block of code
+   */
+  chromeTitle?: string;
+
+  /**
+   * When true, allows the code block to be copied to the user's clipboard by clicking the rendered copy button.
+   *
+   * default: `true`
+   */
+  copyable?: boolean;
+
+  /**
+   * Callback fired when Code is copied via the copy button.
+   *
+   */
+  onCopy?: Function;
```

Comment:
We can still probably be more specific here

Reply:
There isn't really more specificity needed here. It's a function invoked with 0 parameters.